app/models/person.rb

```diff
@@ -66,4 +66,14 @@ def for_feed
   def latest_person_escort_record
     person_escort_records.where(status: 'confirmed').order(confirmed_at: :desc).first
   end
+
+  def category
+    if latest_nomis_booking_id.present?
+      @category ||= begin
+                      booking_details = NomisClient::BookingDetails.get(latest_nomis_booking_id)
```

Comment:
We currently rescue timeout and connection errors, however if NOMIS returns an internal error (surfaced as an oauth error) here the whole step would fail, should we instead gracefully rescue that error with the correct messaging for clients?

Reply:
I'm not entirely sure how `OAuth2::Error` Nomis errors should generally be handled... but looking at the court bookings endpoint I've added code to log this to Sentry and then render it as a 504 Bad Gateway error. Is that what you had in mind?

Reply:
Yes that perfect,  we should start adopting this pattern anywhere we might be surfacing 500s if NOMIS fails